### PR TITLE
Node docs last third

### PR DESCRIPTION
**Pull request by IskyN**
Documented the first 6 of my 20 astroid nodes to document. (The Dict node is to be documented by all 3 ROPes, but my version should not differ from the others'.)
Documented nodes:
- Dict (test node, documented by everyone)
- List
- ListComp
- Name
- Nonlocal
- Pass

Edit 2016-10-02: Now documented, with better template, until Slice.py.

### CODE REVIEW

File: nodes/SetComp.py [hunk 1]
```diff
@@ -1 +1,30 @@
...
+
+This node represents set comprehension expressions in Python, which take the
+form {element for increment in iterable}. This node also makes use of the
+Comprehension node.
+
+Attributes:
+    - elt         (Expr)
+        - The part to be evaluated to make every item in the set. This can
+          be any expression node, such as a BinOp or Str.
+    - generators  (List[])
```
Comment (david-yz-liu) [inline nodes/SetComp.py]: In general for the `List` type, either you don't specify the type of the contents and just write `List`, or you *do* specify the type of the contents, e.g. `List[Comprehension]`.

File: nodes/Raise.py [hunk 2]
```diff
@@ -1 +1,47 @@
-raise ZeroDivisionError()
+"""
+Raise astroid node
+
+This node represents the "raise" statement in Python, which is used to raise
+exceptions. This statement also handles exception chaining using the "from"
+clause, which is included in the Raise node.
+
+Attributes:
+    - exc    (node)
```
Comment (david-yz-liu) [inline nodes/Raise.py]: Make sure to capitalize `Node` when referring to it as a class name. Also, express the fact that this attribute could be none in its type: `Node | None`

File: nodes/List.py [hunk 3]
```diff
@@ -1 +1,25 @@
+"""
+List astroid node
```
Comment (david-yz-liu) [inline nodes/List.py]: We talked about trying to get a "Store" context for lists. I wonder if this is achieved by the assignment statement

```python
[x, y] = [1, 2]
```

File: nodes/Raise.py (hunk 2 above)
Comment (IskyN) [inline nodes/Raise.py]: Got it, thanks.

File: nodes/List.py (hunk 3 above)
Comment (IskyN) [inline nodes/List.py]: I checked, and yes it is! Thank you for that :)

File: nodes/SetComp.py (hunk 1 above)
Comment (IskyN) [inline nodes/SetComp.py]: Oops, I just got distracted before finishing that one, thanks for pointing it out.

**Pull request closed without merging**